plugins/scenario/hu.bme.mit.gamma.scenario.statechart.generator/src/hu/bme/mit/gamma/scenario/statechart/generator/MonitorStatechartGenerator.xtend

```diff
@@ -44,6 +47,8 @@ class MonitorStatechartGenerator extends AbstractContractStatechartGeneration {
 	protected List<Pair<StateNode, StateNode>> copyOutgoingTransitionsForOpt = newLinkedList
 	protected boolean restartOnColdViolation = false
 
+	protected boolean useColdViolationForEnvViolation = true
```

Comment:
Maybe I am missing something, but if we have such a variable for environment violation, why is there not one for the component violation? Also, these configuration variables should be final.